llvm/include/llvm/DebugInfo/DWARF/DWARFAcceleratorTable.h

```diff
@@ -543,6 +543,27 @@ class DWARFDebugNames : public DWARFAcceleratorTable {
       return StrData.getCStr(&Off);
     }
 
+    // Compares the name of this entry against Target, returning true if they
+    // are equal. This is helpful is hot code paths that do not need the length
+    // of the name.
+    bool compareNameAgainst(StringRef Target) const {
+      // Note: this is not the name, but the rest of debug_names starting from
+      // name. This handles corrupt data (non-null terminated) without
+      // overruning the buffer.
+      auto Data = StrData.getData().substr(StringOffset);
+      auto DataSize = Data.size();
+      auto TargetSize = Target.size();
+
+      // Invariant: at the start of the loop, we have non-null characters to
+      // read from Data.
+      size_t Idx = 0;
+      for (; Idx < DataSize && Data[Idx]; Idx++) {
```

Comment:
Would the code become shorter / more readable / efficient if we used StringRef.equals() here? https://llvm.org/doxygen/StringRef_8h_source.html#l00164
Or is this also not possible?

Reply:
I can anwser my own question: because it would ignore a NUL in `Data`.

Reply:
Note that the first thing `StringRef.equals` does it to compare the lengths of the two StringRefs.
But `Data.size()` is the length of the debug_str section (starting at the strp offset), _not_ the length of the string we are interested in

Reply:
FWIW, it might not hurt to include the (shortened) rationale about str(n)cmp in the code as a comment.

Reply:
In fact we don't know the length of the string we are interested in

Reply:
I suppose the primary reason why `strcmp` isn't faster is because in 99% of the cases this look will fail the comparison very early?